src/main/java/maksym/fedorenko/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -31,7 +32,14 @@ public List<Book> findAll() {
                     session -> session.createQuery("from Book", Book.class).getResultList()
             );
         } catch (Exception e) {
-            throw new BookNotFoundException("Can't load list of books", e);
+            throw new EntityNotFoundException("Can't load list of books", e);
         }
     }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        return sessionFactory.fromSession(
+                session -> Optional.ofNullable(session.find(Book.class, id))
```

Comment:
Why whithout try/catch?) (Maybe that's what I intended, anyway, I have to ask =) )